packages/ods/src/components/pagination/src/components/ods-pagination/ods-pagination.tsx

```diff
@@ -0,0 +1,396 @@
+              return;
+            }
+
+            if (isLeft) {
+              this.handlePreviousClick(Number(this.current));
+            } else {
+              this.handleNextClick(Number(this.current));
+            }
+          }}
+          onKeyUp={(event: KeyboardEvent) => {
+            if (isLeft) {
+              this.handlePreviousKeyUp(event, Number(this.current));
+            } else {
+              this.handleNextKeyUp(event, Number(this.current));
+            }
+          }}
+          variant={ODS_BUTTON_VARIANT.ghost}
+          size={ODS_BUTTON_SIZE.md}
+        >
+        </ods-button>
+        {tooltipLabel && (
+          <ods-tooltip
+            shadowDomTriggerId={arrowButtonId}
+            triggerId={this.hostId}
+          >
+            {tooltipLabel}
+          </ods-tooltip>
+        )}
+      </li>
+    );
+  }
+
+  private renderEllipsis(key: string): typeof Fragment {
+    return (
+      <li key={key}>
+        <ods-button
+          class="ods-pagination__list__page__ellipsis"
+          color={ODS_BUTTON_COLOR.primary}
+          isDisabled={true}
+          label="&#x2026;"
+          variant={ODS_BUTTON_VARIANT.ghost}
+        >
+        </ods-button>
+      </li>
+    );
+  }
+
+  render(): FunctionalComponent | undefined {
+    if (!this.totalItems && this.actualTotalPages < 2) {
+      return;
+    }
+
+    const renderEllipsisLeft = this.current > this.ellipsisThreshold && this.actualTotalPages > this.maxVisibleItems;
+    const renderEllipsisRight = this.current < this.actualTotalPages - this.ellipsisThreshold && this.actualTotalPages > this.maxVisibleItems;
+
+    return (
+      <Host
+        class="ods-pagination"
+        isDisabled={this.isDisabled}
+        id={this.hostId}
+      >
+        {
+          !!this.totalItems &&
+            <div class={{
+              'ods-pagination__results': true,
+              'ods-pagination__results--disabled': this.isDisabled,
+            }}>
+              {
+                this.totalItems >= ODS_PAGINATION_PER_PAGE.option_10 &&
+                  <ods-select isDisabled={this.isDisabled}
+                    value={`${this.itemPerPage}`}
+                    name="ods-pagination__items-per-page"
+                  >
+                    {
+                      ODS_PAGINATION_PER_PAGE_OPTIONS.map((option) => (
+                        <option key={option} value={option}>{option}</option>
+                      ))
+                    }
+                  </ods-select>
+              }
+              <ods-text preset={ODS_TEXT_PRESET.label}>
+                <slot name="before-total-items"></slot>
+                {this.totalItems}
+                <slot name="after-total-items"></slot>
+              </ods-text>
+            </div>
+        }
+
+        <ul class="ods-pagination__list">
+          {this.renderArrow('left')}
+
+          {(this.totalItems || this.actualTotalPages > 1) && (
+            <li key={1}>
```

Comment:
this part can be mutualize in a function there are just the `key` and the `ods-pagination__list__page__button--visible` class who change